DataFormats/Headers/test/testDataHeader.cxx

```diff
@@ -261,10 +261,11 @@ BOOST_AUTO_TEST_CASE(headerStack_test)
   BOOST_CHECK(h1 != nullptr);
   BOOST_CHECK(*h1 == dh1);
   BOOST_CHECK(h1->flagsNextHeader == 1);
-  const NameHeader<0>* h2 = get<NameHeader<0>*>(s1.data());
+  BOOST_CHECK_THROW(get<NameHeader<0>*>(s1.data()), std::runtime_error);
```

Comment:
It turns out that we do not allow this kind of variable size header anymore, if we check for the header size.
The idea of this is that one can retrieve an "any-size" header, while adding the header with the specific size.
So it's probably better to only check the version number and do not require a specific size.

Reply:
not sure what you mean by that? What do we not allow anymore?

Reply:
the original line `get<Nameheader<0>*>` will throw now, because `NameHeader<0>` has a different size than `NameHeader<9>`. That's why the change in the check.

Reply:
it sort of defies the purpose of this particular thing, doesn't it? In this case we do not necessarily know at compile time what the length is going to be...

Reply:
variable length headers need to be supported (was one of the design goals - e.g. for other types of metadata like serialized stuff and so on, think ROOT streamer or something). I think the sanity check has to be extended.

Reply:
I think the check can not be extended, because we do not have the exact type for comparison. So that said, only header version can be checked. I will change the PR.